os: Cannot overwrite subdirectories with Cp

If any subdirectory from `src` exists in `dst/src` os.Cp will fail.

```
src
 ├─ dir
 │  └─ other-file.txt
 └─ file.txt
```

```
dst
 └─ src
    └─ dir
       └─ new-file.txt
```

```go
package main

import (
	"github.com/ntrrg/ntgo/os"
)

func main() {
	if err := os.Cp("dst", "src"); err != nil {
		panic(err)
	}
}
```

```shell-session
$ go run main.go
panic: mkdir dst/src: file exists

goroutine 1 [running]:
main.main()
        /home/ntrrg/go-test/main.go:9 +0x9b
        exit status 2
```

--- a/os/file.go
+++ b/os/file.go
@@ -114,7 +114,7 @@ func copyDir(dst, src string, sfi os.FileInfo) error {
 		return fmt.Errorf("cannot copy directory %v into itself (%s)", src, dst)
 	}
 
-	if err := os.Mkdir(dst, sfi.Mode()); err != nil {
+	if err := os.MkdirAll(dst, sfi.Mode()); err != nil {
 		return err
 	}
 
@@ -133,7 +133,7 @@ func copyDir(dst, src string, sfi os.FileInfo) error {
 		dst := filepath.Clean(strings.Replace(srcpath, src, dst, 1))
 
 		if fi.IsDir() {
-			if err := os.Mkdir(dst, fi.Mode()); err != nil {
+			if err := os.MkdirAll(dst, fi.Mode()); err != nil {
 				errs = append(errs, err.Error())
 				return nil
 			}

--- a/os/file_test.go
+++ b/os/file_test.go
@@ -263,8 +263,12 @@ func TestCp_directoryToDirectory(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	if err := os.MkdirAll(filepath.Join(dst, filepath.Base(src)), 0744); err != nil {
+		t.Fatal(err)
+	}
+
 	if err := ntos.Cp(dst, src); err != nil {
-		t.Errorf("ntos.Cp can't copy %s/ to %s/: %v", src, dst, err)
+		t.Fatalf("ntos.Cp can't copy %s/ to %s/: %v", src, dst, err)
 	}
 
 	dst = filepath.Join(dst, filepath.Base(src))

